process/throttle/antiflood/floodPreventers/quotaFloodPreventer.go

```diff
@@ -121,14 +127,52 @@ func NewQuotaFloodPreventer(arg ArgQuotaFloodPreventer) (*quotaFloodPreventer, e
 // We need the mutOperation here as the get and put should be done atomically.
 // Otherwise we might yield a slightly higher number of false valid increments
 // This method also checks the global sum quota but does not increment its values
-func (qfp *quotaFloodPreventer) IncreaseLoad(pid core.PeerID, size uint64) error {
+func (qfp *quotaFloodPreventer) IncreaseLoad(fromConnectedPid core.PeerID, originator core.PeerID, size uint64) error {
 	qfp.mutOperation.Lock()
 	defer qfp.mutOperation.Unlock()
 
-	return qfp.increaseLoad(pid, size)
+	if originator != qfp.selfPid {
+		return qfp.increaseLoadExternalMessage(fromConnectedPid, originator, size)
+	}
+
+	return qfp.increaseLoadSelfMessage(size)
+}
+
+// increaseLoadExternalMessage is called by a message broadcast by other peer and this call will increase the values
+// for the message originator, the peer that broadcast it to self (if different from the originator) and for the self peer id.
+// The purpose of incrementing also the self peer id is that, in case of multiple mild-flooding connected peers, the current
+// node should not propagate all messages because it will get blacklisted by other peers (it will become a proxy for the flooders)
+// This call will increase the load for current peer taking into account the reserved percentage as it is an external message
+// and not a message sent by self
+func (qfp *quotaFloodPreventer) increaseLoadExternalMessage(fromConnectedPid core.PeerID, originator core.PeerID, size uint64) error {
+	errOriginator := qfp.increaseLoad(originator, size, qfp.percentReserved)
+	var errFromConnected error
```

Comment:
Define above both (as siblings) for better readability / optional.

```
var errFromConnected, errOriginator error
```

Reply:
done, included errSelf